src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -1586,10 +1588,25 @@ def new_nwm_q0(run_results):
                 debuglevel,
             )
 
-            #q0 = run_results
-            q0 = new_nwm_q0(run_results)
+        loop.run_until_complete(asyncio.gather(execute_model, build_next_forcing))
+        run_results = execute_model.result()  # gives us whatever is returned, once it is done.
+        qlats, usgs_df = build_next_forcing.result()  # gives us whatever is returned, once it is done.
+
+        #loop.run_until_complete(build_warm_state)
+        #loop.run: output
+        # TODO: This may create the last forcing twice -- Fix it.
+        if run_set_iterator < len(run_sets) - 1:
+            loop.run_until_complete(asyncio.gather(execute_model, build_next_forcing))
+            run_results = execute_model.result()  # gives us whatever is returned, once it is done.
+            qlats, usgs_df = build_next_forcing.result()  # gives us whatever is returned, once it is done.
+            new_q0 = loop.run_in_executor(None, new_nwm_q0, run_results)
+            loop.run_until_complete(asyncio.gather(new_q0))
+            q0 = new_q0.result()
+        else:
+            loop.run_until_complete(asyncio.gather(execute_model))
+            run_results = execute_model.result()  # gives us whatever is returned, once it is done.
 
-        nwm_output_generator(
+        output_last_run = loop.run_in_executor(None, nwm_output_generator,
```

Comment:
If writing is to distinct files, then asynchronous is no problem. If writing is to the same outputfile (i.e. appending) then we have a race condition here.